src/core/Akka/Actor/ActorCell.cs

```diff
@@ -77,90 +74,113 @@ private LatestFirstSystemMessageList UnstashAll()
         public ActorCell(ActorSystemImpl system, IInternalActorRef self, Props props, MessageDispatcher dispatcher, IInternalActorRef parent)
         {
             _self = self;
-            _props = props;
-            _systemImpl = system;
+            Props = props;
+            SystemImpl = system;
             Parent = parent;
             Dispatcher = dispatcher;
-
         }
 
         /// <summary>
-        /// TBD
+        /// The current message the actor is processing right now.
         /// </summary>
-        public object CurrentMessage { get; internal set; }
+        /// <remarks>
+        /// Will be set to <c>null</c> when not processing any message.
+        /// </remarks>
+        public object? CurrentMessage { get; internal set; }
+        
         /// <summary>
-        /// TBD
+        /// This actor's mailbox instance.
         /// </summary>
-        public Mailbox Mailbox => _mailboxDoNotCallMeDirectly;
+        public Mailbox? Mailbox => _mailboxDoNotCallMeDirectly;
 
         /// <summary>
-        /// TBD
+        /// This actor's message dispatcher.
         /// </summary>
         public MessageDispatcher Dispatcher { get; private set; }
+        
         /// <summary>
-        /// TBD
+        /// Returns <c>true</c> if this actor is local to the system. Can only return <c>false</c> if this actor
+        /// is remotely deployed from elsewhere.
         /// </summary>
         public bool IsLocal { get { return true; } }
+        
         /// <summary>
-        /// TBD
+        /// A reference to the current actor instance
         /// </summary>
-        internal ActorBase Actor { get { return _actor; } }
+        internal ActorBase? Actor { get; private set; }
```

Comment:
Technically needs to be nullable because this field will briefly be `null` during restarts and before `ActorCell.Init` is called.  Has always been this way.

Also, refactored it to use an automatic backing field.